src/pages/resource/ContributorsPanel.tsx

```diff
@@ -1,27 +1,45 @@
-import React, { useReducer } from 'react';
-import { useTranslation } from 'react-i18next';
+import React, { useReducer, useState } from 'react';
 import { useSelector } from 'react-redux';
 
 import mockContributors from '../../utils/testfiles/contributors.json';
+import { USE_MOCK_DATA } from '../../utils/constants';
 
 import TabPanel from '../../components/TabPanel/TabPanel';
 import { contributorReducer } from '../../redux/reducers/contributorReducer';
 import { RootStore } from '../../redux/reducers/rootReducer';
-import Contributor from './contributors/Contributor';
-import ContributorLabel from './contributors/ContributorLabel';
-import ContributorValidator from './contributors/ContributorValidator';
-import StyledContributor from './contributors/StyledComponents';
+import Contributors from './contributors/Contributors';
+import { addContributor } from '../../redux/actions/contributorActions';
+import { emptyContributor } from './../../types/contributor.types';
+import OtherContributors from './contributors/OtherContributors';
 
 interface ContributorsPanelProps {
   onClick: (event: React.MouseEvent<any>) => void;
   tabNumber: number;
 }
 
 const ContributorsPanel: React.FC<ContributorsPanelProps> = ({ onClick, tabNumber }) => {
-  const { t } = useTranslation();
   const errors = useSelector((store: RootStore) => store.errors);
-  const initialState = mockContributors;
+  const initialState = USE_MOCK_DATA ? mockContributors : [];
   const [contributors, dispatch] = useReducer(contributorReducer, initialState);
+  const [idCounter, setIdCounter] = useState(contributors.length);
+
+  const onAddOtherContributor = () => {
+    setIdCounter(idCounter + 1);
+    const newContributor = {
+      ...emptyContributor,
+      ...{ id: 'c' + idCounter, type: '' },
+    };
+    dispatch(addContributor(newContributor));
+  };
+
+  const onAddAuthor = () => {
```

Comment:
technically burde denne hete handleAddAuthor. Samme for handleAddOtherContributor